libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Solo.cpp

```diff
@@ -78,18 +79,34 @@ void AP_BattMonitor_SMBus_Solo::timer()
     // read the button press indicator
     if (read_block(BATTMONITOR_SMBUS_MANUFACTURE_DATA, buff, 6, false) == 6) {
         bool pressed = (buff[1] >> 3) & 0x01;
-
+        static bool power_off_started = false;
+        
         if (_button_press_count >= BATTMONITOR_SMBUS_SOLO_BUTTON_DEBOUNCE) {
             // battery will power off
+            power_off_started = true;
             AP_Notify::flags.powering_off = true;
+            
+            // send a mavlink message too all components if the system is about to power off
+            if(!_state.is_powering_off && power_off_started) {
+                _state.is_powering_off = true;
+                // create command long mavlink message
+                mavlink_message_t msg;
+                mavlink_command_long_t cmd_msg;
+                cmd_msg.command = MAV_CMD_POWER_OFF_INITIATED;
+                mavlink_msg_command_long_encode(mavlink_system.sysid, MAV_COMP_ID_ALL, &msg, &cmd_msg);
+                // send to all vehicle components and GCS
+                GCS_MAVLINK::send_to_components(&msg);
+                gcs().send_text(MAV_SEVERITY_WARNING, "Vehicle powering off");
```

Comment:
There is a race condition on send texts when FrSky telem is enabled, which this could hit. However that is a preexsisting bug (that needs to be fixed soon), and it strikes me as unlikely that someone has both a Solo battery and FrSky telem. You could move this further out on the battery read() method if you wanted, or possibly extract the MAVLink generation out to some common GCS code which monitors the notify flag. (The main advantage of this is other systems could sett he flag in the future and would be consistently handled from the GCS).

Reply:
> it strikes me as unlikely that someone has both a Solo battery and FrSky telem

Not sure that is true, there are other vehicles using Solo batteries - namely @AionRobotics ones.

If this is a pre-existing bug it shouldn't hold up the PR though. Is there an open issue for this?